protocols/gossipsub/src/behaviour.rs

```diff
@@ -576,15 +571,10 @@ where
         // announce to all peers
         let peer_list = self.peer_topics.keys().cloned().collect::<Vec<_>>();
         if !peer_list.is_empty() {
```

Comment:
It is not related to this PR but what is the point of this check? A loop with no entries is a no-op so we don't need to check this, right?

If you don't mind a bit of clean-up while we are at it, you could send a separate PR for tiny refactorings like this.

Note that as is, the code does have a slight performance optimisation in that we don't allocate if we the list is empty. But we can alleviate that be inlining the construction of the event into the loop and thus never perform it when the list is empty.

Reply:
yup makes sense Thomas, did it in f2c5edd766f5ad0153ef292b7e150013ddfb970b